Solaris-specific support for the SolarisIA extension

Not suitable for upstream

--- a/Makefile.am	2019-02-12 11:06:48.069252402 -0800
+++ b/Makefile.am	2019-02-12 11:07:02.819680328 -0800
@@ -58,6 +58,7 @@
 	$(GLX_DIR) \
 	$(PRESENT_DIR) \
 	$(DRI3_DIR) \
+	IA \
 	exa \
 	$(GLAMOR_DIR) \
 	config \
--- a/configure.ac	2019-02-12 11:07:15.756272730 -0800
+++ b/configure.ac	2019-02-12 11:08:35.022110532 -0800
@@ -1394,6 +1394,10 @@
 XI_LIB='$(top_builddir)/Xi/libXi.la'
 XI_INC='-I$(top_srcdir)/Xi'
 
+# SolarisIA extension
+IA_LIB='$(top_builddir)/IA/libIA.la'
+XEXT_LIB="${XEXT_LIB} ${IA_LIB}"
+
 AM_CONDITIONAL(XF86UTILS, test "x$XF86UTILS" = xyes)
 AM_CONDITIONAL(VGAHW, test "x$VGAHW" = xyes)
 AM_CONDITIONAL(VBE, test "x$VBE" = xyes)
@@ -2593,3 +2597,6 @@
 xorg-server.pc
 ])
 AC_OUTPUT
+
+# Add Sun IA extension
+AC_OUTPUT([IA/Makefile])
--- a/include/extinit.h	2019-02-12 11:09:17.830801151 -0800
+++ b/include/extinit.h	2019-02-12 11:09:41.001574924 -0800
@@ -115,6 +115,14 @@
 extern void ShmExtensionInit(void);
 #endif
 
+#define SolarisIAExtension
+
+#ifdef SolarisIAExtension
+#include <X11/extensions/interactive.h>
+extern _X_EXPORT Bool noIAExtension;
+extern void IAExtensionInit(void);
+#endif
+
 extern void SyncExtensionInit(void);
 
 extern void XCMiscExtensionInit(void);
--- a/mi/miinitext.c	2019-02-12 11:09:58.055428427 -0800
+++ b/mi/miinitext.c	2019-02-12 11:14:03.884738651 -0800
@@ -174,6 +174,9 @@
     {XvExtensionInit, "XVideo", &noXvExtension},
     {XvMCExtensionInit, "XVideo-MotionCompensation", &noXvExtension},
 #endif
+#ifdef SolarisIAExtension
+    {IAExtensionInit, "SolarisIA", &noIAExtension},
+#endif
 #ifdef XSELINUX
     {SELinuxExtensionInit, "SELinux", &noSELinuxExtension},
 #endif
--- a/os/utils.c	2019-02-12 11:14:17.355625198 -0800
+++ b/os/utils.c	2019-02-12 11:14:31.122468554 -0800
@@ -173,6 +173,9 @@
 #ifdef DRI2
 Bool noDRI2Extension = FALSE;
 #endif
+#ifdef SolarisIAExtension
+Bool noIAExtension = FALSE;
+#endif
 
 Bool noGEExtension = FALSE;
 
